modules/arrow/src/geoarrow/convert-geojson-feature-to-arrow.ts

```diff
@@ -0,0 +1,319 @@
+import {
+  Float64 as ArrowFloat,
+  FixedSizeList,
+  Field as ArrowField,
+  List as ArrowList,
+  Vector as ArrowVector,
+  makeBuilder
+} from 'apache-arrow';
+import {
+  Point,
+  MultiPoint,
+  LineString,
+  MultiLineString,
+  Polygon,
+  MultiPolygon,
+  Feature
+} from 'geojson';
+
+/**
+ * Arrow Field and geometry vector
+ */
+export type GeoArrowReturnType = {
+  field: ArrowField;
+  geometry: ArrowVector;
+};
+
+/**
+ * Convert GeoJson Features to GeoArrow
+ * @param name
+ * @param features
+ * @returns GeoArrowReturnType
+ */
+export function geojsonFeaturesToArrow(name: string, features: Feature[]): GeoArrowReturnType {
+  // get the geometry type by iterating all features
+  // if one geometry is multi-, then the whole geometry is multi-
+  let geometryType = '';
+  for (let i = 0; i < features.length; i++) {
+    const {geometry} = features[i];
+    if (i === 0) {
+      geometryType = geometry.type;
+    }
+    if (geometry.type.includes('Multi')) {
+      geometryType = geometry.type;
+      break;
+    }
+  }
+  if (geometryType === 'Point') {
+    return geojsonPointToArrow(name, features);
+  } else if (geometryType === 'MultiPoint') {
+    return geojsonMultiPointToArrow(name, features);
+  } else if (geometryType === 'LineString') {
+    return geojsonLineStringToArrow(name, features);
+  } else if (geometryType === 'MultiLineString') {
+    return geojsonMultiLineStringToArrow(name, features);
+  } else if (geometryType === 'Polygon') {
+    return geojsonPolygonToArrow(name, features);
+  } else if (geometryType === 'MultiPolygon') {
+    return geojsonMultiPolygonToArrow(name, features);
+  }
+  throw new Error('Unsupported geometry type');
+}
+
+/**
+ * convert GeoJSON Point to geoarrow
+ * @param name geometry field name
+ * @param points GeoJSON Points
+ * @returns GeoArrowReturnType
+ */
+export function geojsonPointToArrow(name: string, points: Feature[]): GeoArrowReturnType {
+  // get dimension from the first point
+  const dimension = (points[0].geometry as Point).coordinates.length;
+  const pointFieldName = dimension === 2 ? 'xy' : 'xyz';
+
+  // get point type
+  const nullable = false;
+  const coordType = new ArrowFloat();
+  const pointType = new FixedSizeList(
+    dimension,
+    new ArrowField(pointFieldName, coordType, nullable)
+  );
+
+  // create a field
+  const metaData: Map<string, string> = new Map([['ARROW:extension:name', 'geoarrow.point']]);
+  const field = new ArrowField(name, pointType, nullable, metaData);
+
+  // make geoarrow builder
+  const builder = makeBuilder({
+    type: pointType,
+    nullValues: [null]
+  });
```

Comment:
Fwiw I've never tried the Arrow JS builders.

Like how the geojson-binary geojson conversion is implemented in loaders, in my own work I try to do a two-pass approach where I first count the number of coordinates and thus infer the sizes of buffers needed, and then in a second step allocate the exact size of buffers once and fill them with coordinates. 

In your case if you wanted to go that route you could count the number of coordinates you'll have, instantiate a single `Float64Array` and set the values with coordinates. For nested types you'd have to manage building your own offset buffers as well, which may be a bit more complex than you want.

Reply:
While your two-pass approach is surely more optimized It would be nice to build a baseline with the builders, as this presumably lets us work with more sanely with complex data types, that can have many index columns.

We can add such optimizations later.